modyn/storage/internal/file_wrapper/csv_file_wrapper.py

```diff
@@ -0,0 +1,198 @@
+            self.ignore_first_line = False
+
+        if "encoding" in file_wrapper_config:
+            self.encoding = file_wrapper_config["encoding"]
+        else:
+            self.encoding = "utf-8"
+
+        # check that the file is actually a CSV
+        self._validate_file_extension()
+
+        self._validate_file_content()
+
+    def _validate_file_extension(self) -> None:
+        """Validates the file extension as csv
+
+        Raises:
+            ValueError: File has wrong file extension
+        """
+        if not self.file_path.endswith(".csv"):
+            raise ValueError("File has wrong file extension.")
+
+    def _validate_file_content(self) -> None:
+        """
+        Performs the following checks:
+        - specified label column is castable to integer
+        - each row has the label_index_column
+        - each row has the same width
+
+        Raises a ValueError if a condition is not met
+        """
+
+        reader = self._get_csv_reader()
+
+        number_of_columns = []
+
+        for row in reader:
+            number_of_columns.append(len(row))
+            if self.label_index is not None:
+                if not 0 <= self.label_index < len(row):
+                    raise ValueError("Label index outside row boundary")
+                if not row[self.label_index].isnumeric():  # returns true iff all the characters are numbers
+                    raise ValueError("The label must be an integer")
+
+        if len(set(number_of_columns)) != 1:
+            raise ValueError(
+                "Some rows have different width. " f"This is the number of columns row by row {number_of_columns}"
+            )
+
+    def get_sample(self, index: int) -> bytes:
+        samples = self._filter_rows_samples([index])
+
+        if len(samples) != 1:
+            raise IndexError("Invalid index")
+
+        return samples[0]
+
+    def get_samples(self, start: int, end: int) -> list[bytes]:
+        indices = list(range(start, end))
+        return self.get_samples_from_indices(indices)
+
+    def get_samples_from_indices(self, indices: list) -> list[bytes]:
+        samples = self._filter_rows_samples(indices)
+
+        if len(samples) != len(indices):
+            raise IndexError("At least one index is invalid.")
+
+        return samples
+
+    def get_label(self, index: int) -> Optional[int]:
+        if self.label_index is None:
+            return None
+
+        labels = self._filter_rows_labels([index])
+
+        if len(labels) != 1:
+            raise IndexError("Invalid index.")
+
+        return labels[0]
+
+    def get_all_labels(self) -> list[Optional[int]]:
+        reader = self._get_csv_reader()
+
+        labels: list[Optional[int]] = []
+
+        if self.label_index is None:
+            return [None] * self.get_number_of_samples()
+
+        for row in reader:
+            # labels are integer in modyn
+            int_label = int(row[self.label_index])
+            labels.append(int_label)
```

Comment:
maybe use list comprehension here to define labels? might avoid resizing several times